_includes/tutorials/session-windows/kstreams/code/src/main/java/io/confluent/developer/SessionWindow.java

```diff
@@ -0,0 +1,215 @@
+
+        final HashMap<String, String> serdeConfig = new HashMap<>();
+        serdeConfig.put(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
+                envProps.getProperty("schema.registry.url"));
+
+        specificAvroSerde.configure(serdeConfig, false);
+        return specificAvroSerde;
+    }
+    public void createTopics(Properties envProps) {
+        Map<String, Object> config = new HashMap<>();
+        config.put("bootstrap.servers", envProps.getProperty("bootstrap.servers"));
+        try (AdminClient client = AdminClient.create(config)) {
+            List<NewTopic> topicList = new ArrayList<>();
+
+            NewTopic sessionInput = new NewTopic(envProps.getProperty("input.topic.name"),
+                    Integer.parseInt(envProps.getProperty("input.topic.partitions")),
+                    Short.parseShort(envProps.getProperty("input.topic.replication.factor")));
+            topicList.add(sessionInput);
+
+            NewTopic counts = new NewTopic(envProps.getProperty("output.topic.name"),
+                    Integer.parseInt(envProps.getProperty("output.topic.partitions")),
+                    Short.parseShort(envProps.getProperty("output.topic.replication.factor")));
+
+            topicList.add(counts);
+            client.createTopics(topicList);
+        }
+    }
+
+    public Properties loadEnvProperties(String fileName) throws IOException {
+        Properties envProps = new Properties();
+        FileInputStream input = new FileInputStream(fileName);
+        envProps.load(input);
+        input.close();
+
+        return envProps;
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        if (args.length < 1) {
+            throw new IllegalArgumentException("This program takes one argument: the path to an environment configuration file.");
+        }
+
+        SessionWindow tw = new SessionWindow();
+        Properties envProps = tw.loadEnvProperties(args[0]);
+        Properties streamProps = tw.buildStreamsProperties(envProps);
+        Topology topology = tw.buildTopology(envProps);
+
+        tw.createTopics(envProps);
+        ClicksDataGenerator dataGenerator = new ClicksDataGenerator(envProps);
+        dataGenerator.generate();
+
+        final KafkaStreams streams = new KafkaStreams(topology, streamProps);
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        // Attach shutdown handler to catch Control-C.
+        Runtime.getRuntime().addShutdownHook(new Thread("streams-shutdown-hook") {
+            @Override
+            public void run() {
+                streams.close();
+                latch.countDown();
+            }
+        });
+
+        try {
+            streams.cleanUp();
+            streams.start();
+            latch.await();
+        } catch (Throwable e) {
+            System.exit(1);
+        }
+        System.exit(0);
+    }
+
+    static class ClicksDataGenerator {
```

Comment:
I included an inner class to generate records for this tutorial.  Usually, I would use the data-gen connector but session windows require setting timestamps with enough time to enforce a new session starting, and this approach makes it easier to create separate "sessions".  I've noted in the tutorial the use of this class and that users shouldn't include this in their code.   
 
IMHO it's a better user experience to have the input topic populate with records automatically vs. having them go through the step themselves.

\cc @gAmUssA

Reply:
i love it. IIUC this datagen is on-off, right? not continuous

Reply:
> is on-off, right? not continuous

Yes, it produces 8 records then quits